cpp/src/arrow/compute/exec/sink_node.cc

```diff
@@ -103,6 +157,28 @@ class SinkNode : public ExecNode {
 
   Future<> finished() override { return finished_; }
 
+  void RecordBackpressureBytesUsed(const ExecBatch& batch) {
+    if (backpressure_queue_->enabled()) {
+      uint64_t bytes_used = static_cast<uint64_t>(batch.TotalBufferSize());
+      auto state_change = backpressure_queue_->RecordProduced(bytes_used);
+      if (state_change >= 0) {
+        EVENT(span_, "Backpressure applied", {{"counter", state_change}});
```

Comment:
nit, but maybe namespace the span attribute name?

Reply:
I changed it to `backpressure.counter`.  Is this what you were thinking?  It doesn't seem we are doing much namespacing elsewhere.

Reply:
That works, I was thinking `arrow....` like in the memory PR but it's not a big deal.